Packs/AzureCompute/Integrations/AzureCompute_v2/AzureCompute_v2.py

```diff
@@ -872,26 +915,79 @@ def get_public_ip_details_command(client: MsGraphClient, args: dict):
         'PublicIPAddressVersion': ip_address_version,
         'PublicIPAddressAllocationMethod': ip_address_allocation_method,
         'PublicIPAddressDomainName': address_domain_name,
-        'PublicIPAddressFQDN': address_fqdn
+        'PublicIPAddressFQDN': address_fqdn,
+        'NIC': nic
     }
 
     human_readable_ip_config = {
         'PublicConfigName': config_name,
         'Location': location,
         'PublicIPAddress': ip_address,
         'PublicIPAddressVersion': ip_address_version,
-        'PublicIPAddressAllocationMethod': ip_address_allocation_method
+        'PublicIPAddressAllocationMethod': ip_address_allocation_method,
+        "ResourceGroup": resource_group,
+        'NIC': nic
     }
 
     title = 'Properties of Public Address "{}"'.format(address_name)
     table_headers = ['PublicConfigName', 'Location', 'PublicIPAddress', 'PublicIPAddressVersion',
-                     'PublicIPAddressAllocationMethod']
+                     'PublicIPAddressAllocationMethod', 'ResourceGroup', 'NIC']
     human_readable = tableToMarkdown(title, human_readable_ip_config, headers=table_headers, removeNull=True)
     entry_context = {'Azure.Network.IPConfigurations(val.PublicIPAddressID === '
                      'obj.PublicIPAddressID)': ip_config}
     return human_readable, entry_context, response
 
 
+def get_all_public_ip_details_command(client: MsGraphClient, args: dict):
+    """
+    Get the properties of all Public IP Addresses in the configured subscription
+
+    returns:
+        List of Public IP Address Objects
+    """
+    response = client.get_all_public_ip_details()
+
+    ip_objects_list = response.get('value')
+    ips = []
+
+    for ip_object in ip_objects_list:
+        # Retrieve relevant properties to return to context
+        properties = ip_object.get('properties', {})
+        address_id = ip_object.get('id')
+        config_id = properties.get('ipConfiguration', {}).get('id')
+        ip_address = properties.get('ipAddress', 'NA')
+        ip_address_version = properties.get('publicIPAddressVersion', 'NA')
+        ip_address_allocation_method = properties.get('publicIPAllocationMethod', 'NA')
+        address_domain_name = properties.get('dnsSettings', {}).get('domainNameLabel', 'NA')
+        address_fqdn = properties.get('dnsSettings', {}).get('fqdn', 'NA')
+        config_name = ip_object.get('name')
+        location = ip_object.get('location')
+        resource_group = address_id.split('resourceGroups/')[1].split('/providers')[0]
+        nic = config_id.split('networkInterfaces/')[1].split('/ipConfigurations')[0]
```

Comment:
Is this always going to have ('networkInterfaces/')[1] otherwise we will face an index error

Reply:
I went ahead and removed this and the new mentions of the NIC for the `yml` and `README` files as well. We decided that we will handle it in a playbook instead for our use case.